monix-eval/shared/src/main/scala/monix/eval/TaskLocal.scala

```diff
@@ -95,6 +95,15 @@ import monix.execution.misc.Local
 final class TaskLocal[A] private (default: => A) {
   private[this] val ref = new Local(default)
 
+  /** Returns [[monix.execution.misc.Local]] instance used in this [[TaskLocal]].
+    *
+    * Note that `TaskLocal.bind` will restore the original local value
+    * on the thread where the `Task's` run-loop ends up so it might lead
+    * to leaving local modified in other thread.
+    */
+  def local: Task[Local[A]] =
+    Task.eval(ref)
```

Comment:
If this is going to be exposed, shouldn't we define read/write/clean with `local`?

```scala
def read: Task[A] =
  local.map(_.get)
```

Reply:
Not necessarily. It's more elegant, but internals are allowed to be ugly for performance reasons 😀